luda-adventure/src/app/game/game.rs

```diff
@@ -0,0 +1,128 @@
+use super::{player_character::new_player, *};
+use namui::prelude::*;
+
+const NEAR_FUTURE: Time = Time::Sec(5.0);
+
+pub struct Game {
+    pub state: GameState,
+    pub camera: Camera,
+    pub ecs_app: crate::ecs::App,
+}
+impl Game {
+    pub fn new() -> Self {
+        let character = mock_character();
+        let quest_object = mock_quest_object();
+        let walls = mock_walls();
+        let floors = mock_floor();
+        let state = GameState::new();
+
+        let player_entity_id = character.id();
+
+        let mut ecs_app = crate::ecs::App::new();
+        ecs_app.add_entity(character);
+        ecs_app.add_entity(quest_object);
+        ecs_app.add_entities(walls);
+        ecs_app.add_entity(floors);
+
+        Self {
+            state,
+            camera: Camera::new(Some(CameraSubject::Object {
+                id: player_entity_id,
+            })),
+            ecs_app,
+        }
+    }
+
+    pub fn update(&mut self, event: &dyn std::any::Any) {
+        let current_time = now();
+        if let Some(event) = event.downcast_ref::<NamuiEvent>() {
+            match event {
+                NamuiEvent::KeyDown(_) => {
+                    self.handle_user_input_for_character_movement(current_time)
+                }
+                NamuiEvent::KeyUp(_) => self.handle_user_input_for_character_movement(current_time),
```

Comment:
Event can be queued so current system's state could be different to last event. So I think you should use event, not system.keyboard. how do you think about it?

Reply:
That makes sense. I will fix it.